### DISCUSSION TITLE
How to select a different GPU for the OpenGL context ?

### DISCUSSION THREAD

**Discussion by sebbeutler** [slint-ui/slint#4987] [Q&A] [answered]
I'm trying to run the [OpenGL texture example](https://github.com/slint-ui/slint/tree/master/examples/opengl_texture) and I noticed the shader version use is `#version 100`. I would like to use a newer version like `#version 330` but when i make the changes I get an error `ERROR: 0:1: '#version' :  version number '5' not supported by current context`.

With some closer inspection, the program is using the Intel UHD gpu with OpenGL ES 2.0.

How can I specify the usage of another gpu, in this case I have an NVidia RTX gpu as well, or use another OpenGL context that handle newer GLSL versions ?

**Comment by tronical** [comment, marked as answer]
At the moment that's not possible, we default to OpenGL ES.

One thing that you could try is to create a separate _sharing_ GL context, render into a texture there, but - due to sharing - be able to use the same texture id with slint::Image.

Long term, it would be really nice to have API that allows users to request certain aspects of the graphics configuration at startup, such as a preference for the renderer, or for example a preference for the OpenGL profile.
